hspec-core/src/Test/Hspec/Core/Runner.hs

```diff
@@ -450,11 +453,11 @@ toEvalItemForest :: Params -> [SpecTree ()] -> [EvalItemTree]
 toEvalItemForest params = bimapForest id toEvalItem . filterForest itemIsFocused
   where
     toEvalItem :: Item () -> EvalItem
-    toEvalItem (Item requirement loc isParallelizable _isFocused e) = EvalItem {
+    toEvalItem (Item requirement loc isParallelizable _isFocused e maxTime) = EvalItem {
       evalItemDescription = requirement
     , evalItemLocation = loc
     , evalItemConcurrency = if isParallelizable == Just True then Concurrent else Sequential
-    , evalItemAction = \ progress -> measure $ e params withUnit progress
+    , evalItemAction = \ progress -> fmap (fmap $ fromMaybe $ Result requirement $ Failure loc $ Reason "exceeded timeout") $ measureWithTimeout maxTime $ e params withUnit progress
```

Comment:
As I understand it, you could use  `maxTime` instead of `configMaxTimePerTest config` here, for now.

Also, that line of code is getting quite long. I think it could make sense to extract some of the code into a separate function, say `applyTimeout` or `failOnTimeout` or both, not sure exactly how it will turn out nice, maybe just try something.

Reply:
> As I understand it, you could use  maxTime instead of configMaxTimePerTest config here, for now.

Sorry, I don't understand this.